cmd/rm-main.go

```diff
@@ -121,14 +132,24 @@ func checkRmSyntax(ctx *cli.Context) {
 	isForce := ctx.Bool("force")
 	isRecursive := ctx.Bool("recursive")
 	isStdin := ctx.Bool("stdin")
-
+	isNamespace := ctx.Bool("namespace")
+	isNamespaceRemoval := false
+	for _, url := range ctx.Args() {
+		if isHostName(url) {
+			isNamespaceRemoval = true
+			break
+		}
+	}
 	if !ctx.Args().Present() && !isStdin {
 		exitCode := 1
 		cli.ShowCommandHelpAndExit(ctx, "rm", exitCode)
 	}
-
+	if (isRecursive || isStdin) && isNamespaceRemoval && !isNamespace {
+		fatalIf(errDummy().Trace(),
+			"Full namespace removal requires --namespace option. This operation removes all the buckets on your namespace. Please review carefully before performing this *DANGEROUS* operation.")
```

Comment:
This check is not interesting.  removing local file system, we don't need any restriction like that.